src/inabox/inabox-viewport.js

```diff
@@ -81,11 +85,21 @@ export class ViewportBindingInabox {
     // 2) broadcast the request
     this.iframeClient_.setHostWindow(win.top);
 
+    /** @private {boolean} */
+    this.visibilityV2Enabled_ =
+        nativeIntersectionObserverSupported(win) &&
+            isExperimentOn(win, 'visibility-v2');
+
     dev().fine(TAG, 'initialized inabox viewport');
   }
 
   /** @override */
   connect() {
+    if (this.visibilityV2Enabled_) {
+      // Visibility V2 uses native IntersectionObserver, no position data needed
```

Comment:
Then I think we can't just return when `this.visiblityV2Enabled_` is true, `connect()` function is not only about amp-analytics visibility tracking, it will also take care of remeasuring viewportBox for runtime use. We can't simply skip that part of logic with `visiblityV2Enabled_` is true?